src/manage_sql_secinfo.c

```diff
@@ -5123,6 +5123,128 @@ manage_sync_scap (sigset_t *sigmask_current)
                 "gvm-sync-scap");
 }
 
+/**
+ * @brief Rebuild part of the SCAP DB.
+ *
+ * @param[in]  type        The type of SCAP info to rebuild.
+ *
+ * @return 0 success, 1 invalid type, -1 error
+ */
+static int
+rebuild_scap (const char *type)
+{
+  int updated_scap_ovaldefs, updated_scap_cpes, updated_scap_cves;
+
+  updated_scap_ovaldefs = 0;
+  updated_scap_cpes = 0;
+  updated_scap_cves = 0;
+
+  if (strcasecmp (type, "ovaldefs") == 0
+      || strcasecmp (type, "ovaldef") == 0)
+    {
+      g_debug ("%s: rebuilding ovaldefs", __FUNCTION__);
+      sql ("DELETE FROM ovalfiles");
+      sql ("DELETE FROM affected_ovaldefs");
+      sql ("DELETE FROM ovaldefs");
+
+      updated_scap_ovaldefs = update_scap_ovaldefs (0, 0 /* Feed data. */);
+      if (updated_scap_ovaldefs == -1)
+        return -1;
+
+      switch (update_scap_ovaldefs (0, 1 /* Private data. */))
```

Comment:
I don't like that these update functions are called here and in sync_scap.  It means there are two places to keep track of what happens during sync, and also that the syncs behave slightly differently.  This version doesn't set proctitle or call manage_update_scap_db_init/cleanup.

Reply:
I've refactored some things in b813f30fbeaefffd42208150f466fdf2b06e7446 to address this.

Reply:
Thank you.  Very neat now.